From bc910686202ff06afa778af5533dec70a1794909 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Mon, 9 May 2022 23:30:24 +0000
Subject: [PATCH 1/2] qat: use default_groups in kobj_type

default_attrs was removed in kernel commit cdb4f26a63c3 ("kobject:
kobj_type: remove default_attrs"), use the default_groups pointer
instead.

Upstream-Status: Inappropriate [Code released in tarball form only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 .../qat/drivers/crypto/qat/qat_common/adf_uio_control.c     | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/quickassist/qat/drivers/crypto/qat/qat_common/adf_uio_control.c b/quickassist/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
index 26cfe58..e653d0b 100644
--- a/quickassist/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
+++ b/quickassist/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
@@ -82,6 +82,7 @@ static struct attribute *accel_default_attrs[] = {
 	&accel_sku.attr,
 	NULL,
 };
+ATTRIBUTE_GROUPS(accel_default);
 
 static ssize_t accel_show(struct kobject *kobj, struct attribute *attr,
 			  char *buf);
@@ -176,6 +177,7 @@ static struct attribute *bundle_default_attrs[] = {
 	&bundle_rings_reserved.attr,
 	NULL,
 };
+ATTRIBUTE_GROUPS(bundle_default);
 
 static ssize_t bundle_show(struct kobject *kobj, struct attribute *attr,
 			   char *buf);
@@ -226,7 +228,7 @@ static void adf_uio_sysfs_cleanup(struct adf_accel_dev *accel_dev)
 static struct kobj_type accel_ktype = {
 	.release = accel_kobject_free,
 	.sysfs_ops = &accel_sysfs_ops,
-	.default_attrs = accel_default_attrs,
+	.default_groups = accel_default_groups,
 };
 
 int adf_uio_sysfs_create(struct adf_accel_dev *accel_dev)
@@ -265,7 +267,7 @@ int adf_uio_sysfs_create(struct adf_accel_dev *accel_dev)
 static struct kobj_type bundle_ktype = {
 	.release = bundle_kobject_free,
 	.sysfs_ops = &bundle_sysfs_ops,
-	.default_attrs = bundle_default_attrs,
+	.default_groups = bundle_default_groups,
 };
 
 int adf_uio_sysfs_bundle_create(struct pci_dev *pdev,
-- 
2.34.1

